tm2/pkg/telemetry/metrics/metrics.go

```diff
@@ -124,6 +128,43 @@ func Init(config config.Config) error {
 		return fmt.Errorf("error parsing exporter endpoint: %s, %w", config.ExporterEndpoint, err)
 	}
 
+	// The exporter embeds a default OpenTelemetry Reader and
+	// implements prometheus.Collector, allowing it to be used as
+	// both a Reader and Collector.
+	promexp, err := prometheus.New()
+	if err != nil {
+		return fmt.Errorf("error creating prometheus exporter, %w", err)
+	}
+
+	providerOptions := []sdkMetric.Option{
+		sdkMetric.WithResource(
+			resource.NewWithAttributes(
+				semconv.SchemaURL,
+				semconv.ServiceNameKey.String(config.ServiceName),
+				// TODO: Get git tag / commit version
+				semconv.ServiceVersionKey.String("1.0.0"),
+				semconv.ServiceInstanceIDKey.String(config.ServiceInstance),
+			),
+		),
+	}
+
+	if config.PrometheusAddr != "" {
+		providerOptions = append(providerOptions, sdkMetric.WithReader(promexp))
+
+		go func() {
+			server := &http.Server{
+				Addr:              config.PrometheusAddr,
+				ReadHeaderTimeout: 5 * time.Second,
+			}
+			http.Handle("/metrics", promhttp.Handler())
```

Comment:
I'm not sure we should be creating a new server and exposing the `/metrics` here

A better place would be the central one we actually use for all endpoints in `tm2/pkg/bft/node` -- you can just expose a func in telemetry to register the handle with the mux (node's mux)

Reply:
FYI: This is what tendermint what doing

Reply:
At first i liked the idea, but i just changed my mind.

Because you want to have the possibility to expose your RPC port to public without publicly expose your metrics

Reply:
Yes it's make total sense, to give you an example, i could run the RPC on `0.0.0.0` for public access and prometheus endpoint on the VPN ip address.